libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,188 @@
+            metrics);
+
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry availabilityTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", availabilityTelemetry.getName());
+            Assert.assertEquals("message", availabilityTelemetry.getData().getMessage());
+            Assert.assertEquals("value", availabilityTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, availabilityTelemetry.getMetrics().get("metric"));
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void trackEventTest() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("hello", "value");
+        Map<String, Double> metrics = new HashMap<>();
+        metrics.put("metric", 0.6);
+        
+        botTelemetryClient.trackEvent("test", properties, metrics);
+
+        Mockito.doAnswer(invocation -> {
+            EventTelemetry eventTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventTelemetry.getName());
+            Assert.assertEquals("value", eventTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, eventTelemetry.getMetrics().get("metric"));
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(EventTelemetry.class));
+    }
+
+    @Test
+    public void trackDependencyTest() {
+        botTelemetryClient.trackDependency(
+                "test", 
+                "target", 
+                "dependencyname", 
+                "data", 
+                OffsetDateTime.now(), 
+                Duration.ofSeconds(1000), // TODO: use computer ticks
+                "result", false);
+
+        Mockito.doAnswer(invocation -> {
+            RemoteDependencyTelemetry remoteDependencyTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", remoteDependencyTelemetry.getType());
+            Assert.assertEquals("target", remoteDependencyTelemetry.getTarget());
+            Assert.assertEquals("dependencyname", remoteDependencyTelemetry.getName());
+            Assert.assertEquals("result", remoteDependencyTelemetry.getResultCode());
+            Assert.assertFalse(remoteDependencyTelemetry.getSuccess());
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(RemoteDependencyTelemetry.class));
+    }
+
+    @Test
+    public void trackExceptionTest() {
+        Exception expectedException = new Exception("test-exception");
+
+        Mockito.doAnswer(invocation -> {
+            ExceptionTelemetry exceptionTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals(expectedException, exceptionTelemetry.getException());
+            Assert.assertEquals("bar", exceptionTelemetry.getProperties().get("foo"));
+            Assert.assertEquals(0.6, exceptionTelemetry.getProperties().get("metric"));
```

Comment:
This is what I saw, here you are not doing the casting, we should unify this behavior

Reply:
get properties is of type String, String hence why the error I mentioned above doesn't happen, I think

Reply:
The problem here is we should not use `getProperties` to get the `metrics`, we should use `getMetrics`, right?